tests/common/helpers/pfc_storm.py

```diff
@@ -0,0 +1,258 @@
+        """
+        self.pfc_queue_idx = q_idx
+
+    def update_peer_info(self, peer_info):
+        """
+        Update the fanout info. Can be invoked after the class init to change the fanout or fanout interface
+        """
+        self._validate_params(expected_args=['peerdevice', 'pfc_fanout_interface'])
+        for key in peer_info:
+            self.peer_info[key] = peer_info[key]
+        if 'hwsku' not in peer_info:
+            self._populate_peer_hwsku()
+        self.update_platform_name()
+        self.peer_device = self.fanout_hosts[self.peer_info['peerdevice']]
+
+    def update_platform_name(self):
+        """
+        Identifies the fanout platform
+        """
+        if 'arista' in self.peer_info['hwsku'].lower():
+            self.platform_name = 'arista'
+        elif 'MLNX-OS' in self.peer_info['hwsku']:
+            self.platform_name = 'mlnx'
+
+    def _update_template_args(self):
+        """
+        Populates all the vars needed by the pfc storm templates
+        """
+        self.extra_vars = dict()
+        self.extra_vars = { "pfc_gen_file": self.pfc_gen_file,
+                       "pfc_queue_index": self.pfc_queue_idx,
+                       "pfc_frames_number": self.pfc_frames_number,
+                       "pfc_fanout_interface": self.peer_info['pfc_fanout_interface'],
+                       "ansible_eth0_ipv4_addr": self.ip_addr,
+                       "peer_hwsku": self.peer_info['hwsku']
+                     }
+        if getattr(self, "pfc_storm_defer_time", None):
+           self.extra_vars.update({"pfc_storm_defer_time": self.pfc_storm_defer_time})
+        if getattr(self, "pfc_storm_stop_defer_time", None):
+           self.extra_vars.update({"pfc_storm_stop_defer_time": self.pfc_storm_stop_defer_time})
+        if getattr(self, "pfc_asym", None):
+           self.extra_vars.update({"pfc_asym": self.pfc_asym})
+
+    def _prepare_start_template(self):
+        """
+        Populates the pfc storm start template
+        """
+        self._update_template_args()
+        self.pfc_start_template = os.path.join(TEMPLATES_DIR, "pfc_storm_{}.j2".format(self.platform_name))
+        self.extra_vars.update({"template_path": self.pfc_start_template})
+
+    def _prepare_stop_template(self):
+        """
+        Populates the pfc storm stop template
+        """
+        self._update_template_args()
+        self.pfc_stop_template = os.path.join(TEMPLATES_DIR, "pfc_storm_stop_{}.j2".format(self.platform_name))
+        self.extra_vars.update({"template_path": self.pfc_stop_template})
+
+    def start_storm(self):
+        """
+        Starts PFC storm on the fanout interfaces
+        """
+        self._prepare_start_template()
+        logger.info("--- Starting PFC storm on {} on interfaces {}"
+                    .format(self.peer_info['peerdevice'], self.peer_info['pfc_fanout_interface']))
+        self.peer_device.exec_template(ANSIBLE_ROOT, RUN_PLAYBOOK, self.inventory, **self.extra_vars)
```

Comment:
Just learnt that the exec_template is invoking ```ansible_playbook``` command. Should this be deprecated and *.yml fiel be converted using to pytest?

Reply:
I had asked this  to Mellanox who added this script. I was told that not all functionality is supported by their fanout config and command modules and they would need this option.
For Arista fanout, I initially had this coded to execute as a bash script on the fanout but later changed to use exec_template so that it can run on any fanout

Reply:
we can have a todo item to change that once we resolve the issue with Mellanox device. A comment in the code would help not to lose track if the history behind such choice.

Reply:
will put in a comment